lib/ansible/modules/cloud/amazon/redshift_iam_roles.py

```diff
@@ -0,0 +1,140 @@
+        required: true
+        choices: ['present', 'absent']
+    roles:
+        description:
+            - "This is a list of IAM roles (Full ARN) to add to the Redshift cluster. Roles attached to the cluster that are not on the list will be removed."
+        required: true
+extends_documentation_fragment:
+    - ec2
+    - aws
+'''
+
+EXAMPLES = '''
+- name: add IAM roles to a redshift cluster
+  redshift_iam_roles:
+    cluster: "staging"
+    state: present
+    roles:
+        - "arn:aws:iam::123456789012:role/superuser"
+        - "arn:aws:iam::123456789012:role/new_hotness"
+
+- name: remove all IAM roles from a redshift cluster
+  redshift_iam_roles:
+    cluster: "staging"
+    state: absent
+'''
+
+RETURN = ''' # '''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import boto3_conn, ec2_argument_spec, get_aws_connection_info, camel_dict_to_snake_dict, HAS_BOTO3
+import traceback
+
+try:
+    from botocore.exceptions import ClientError, BotoCoreError
+except ImportError:
+    pass  # will be detected by imported HAS_BOTO3
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            cluster=dict(type='str', required=True),
+            state=dict(type='str', required=True, choices=['present', 'absent']),
+            roles=dict(type='list', required=True)
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True
+    )
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 required for this module')
+
+    target_cluster = module.params.get('cluster')
+    desired_state = module.params.get('state')
+    target_roles = module.params.get('roles')
+
+    result = {
+        'changed': False
+    }
+
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+    conn = boto3_conn(module, conn_type='client', resource='redshift', region=region, endpoint=ec2_url, **aws_connect_kwargs)
+
+    cluster_roles = conn.describe_clusters(ClusterIdentifier=target_cluster)['Clusters'][0]['IamRoles']
+    current_roles = [x['IamRoleArn'] for x in cluster_roles]
+
+    if desired_state == 'present':
+        roles_to_add = list(set(target_roles) - set(current_roles))
+        roles_to_remove = list(set(current_roles) - set(target_roles))
+        if not roles_to_add:
+            module.exit_json(**result)
+        try:
+            response = conn.modify_cluster_iam_roles(
+                ClusterIdentifier=target_cluster,
+                AddIamRoles=roles_to_add,
+                RemoveIamRoles=roles_to_remove
+            )
+            result['changed'] = True
+        except Exception as e:
+            module.fail_json(msg="Unable to modify IAM role(s): {0}".format(e),
+                             exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+
+    if desired_state == 'absent':
+        if not current_roles:
```

Comment:
When I was reading the docs I entirely misunderstood that this would happen, thinking state=absent would use the `roles` option as well. `roles` is also a required=True option.